py/desisim/simexp.py

```diff
@@ -505,18 +511,31 @@ def simulate_spectra(wave, flux, fibermap=None, obsconditions=None,
         source_fraction[elgs,0]=1.   # ELG are disk only
         source_fraction[lrgs,1]=1.   # LRG are bulge only
         source_fraction[bgss,0]=0.52 # disk comp in BGS
-        source_fraction[bgss,1]=0.48 # bulge comp in BGS
-        
+        source_fraction[bgss,1]=0.48 # bulge comp in BGS       
 
         # source_half_light_radius[:,0] is the half light radius in arcsec for the DISK profile
         # source_half_light_radius[:,1] is the half light radius in arcsec for the BULGE profile        
         # see specsim.GalsimFiberlossCalculator.create_source routine
         source_half_light_radius=np.zeros((nspec,2))
         source_half_light_radius[elgs,0]=0.45 # ELG are disk only, arcsec
         source_half_light_radius[lrgs,1]=1.   # LRG are bulge only, arcsec
-        source_half_light_radius[bgss,0]=4.7  # disk comp in BGS, arcsec
-        source_half_light_radius[bgss,1]=1.3  # bulge comp in BGS, arcsec
-       
+
+        # 4.7 is angular size of z=0.1 disk, and 1.3 is angular size of z=0.1 bulge
+        bgs_disk_z01 = 4.7  # in arcsec
+        bgs_bulge_z01 = 1.3 # in arcsec
+        
+        # Convert to angular size of the objects in this sample with given redshifts
+        if redshift is None:
+            angscales = np.ones(np.sum(bgss))
+        else:
+            bgs_redshifts = redshift[bgss]
+            # Avoid infinities
+            if np.any(bgs_redshifts == 0.):
+                bgs_redshifts[bgs_redshifts==0.] = 0.0001
+            angscales = ( ang_diam_dist(0.1) / ang_diam_dist(bgs_redshifts) ).value
+        source_half_light_radius[bgss,0]= bgs_disk_z01 * angscales # disk comp in BGS, arcsec
+        source_half_light_radius[bgss,1]= bgs_bulge_z01 * angscales  # bulge comp in BGS, arcsec
+        
```

Comment:
What am I misunderstanding?  If `bgs_disk_z01` is the half-light radius at z=0.1 shouldn't you multiply by `(ang_diam_dist(bgs_redshifts) /  ang_diam_dist(0.1) ).value` rather than `( ang_diam_dist(0.1) / ang_diam_dist(bgs_redshifts) ).value` as you have here?

Reply:
The relevant formula is:   angular_size = physical_size / ang_diam_dist
bgs_disk_z01 is the angular size at z=0.1

Therefore:
 physical_size = bgs_disk_z01 * dA(z=0.1)

And finally:
 angular_size_truez = physical_size / dA(z=truez)
                                =  bgs_disk_z01 * ( dA(z=0.1) / dA(z=truez) )

Reply:
Sorry, I don't use markdown so the above looks terrible. The confusion is that bgs_disk_z01 is an _angularsize_ not the actual extent of the object physically. Otherwise you would be correct